actions/src/main/java/com/redhat/service/bridge/actions/ActionProviderFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.redhat.service.bridge.actions;
+
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class ActionProviderFactory {
+
+    @Inject
+    Instance<ActionProvider> actionProviders;
+
+    public ActionProvider getActionProvider(String actionType) {
+        Optional<ActionProvider> ap = actionProviders.stream().filter((a) -> a.getType().equals(actionType)).findFirst();
```

Comment:
In this case this becomes (please check if correct because I'm changing the code directly here on GitHub
```suggestion
        Optional<ActionProvider> ap = actionProviders.stream().filter(ActionProvider::accept).findFirst();
```